packages/migrate/migrations/svelte-4/index.js

```diff
@@ -0,0 +1,73 @@
+import colors from 'kleur';
+import fs from 'node:fs';
+import path from 'node:path';
+import { pathToFileURL } from 'node:url';
+import prompts from 'prompts';
+import glob from 'tiny-glob/sync.js';
+import { bail, check_git } from '../../utils.js';
+import { update_js_file, update_svelte_file } from './migrate.js';
+
+export async function migrate() {
+	if (!fs.existsSync('package.json')) {
+		bail('Please re-run this script in a directory with a package.json');
+	}
+
+	console.log(colors.bold().yellow('\nThis will update files in the current directory\n'));
+
+	const use_git = check_git();
+
+	const response = await prompts({
+		type: 'confirm',
+		name: 'value',
+		message: 'Continue?',
+		initial: false
+	});
+
+	if (!response.value) {
+		process.exit(1);
+	}
+
+	const { default: config } = fs.existsSync('svelte.config.js')
+		? await import(pathToFileURL(path.resolve('svelte.config.js')).href)
+		: { default: {} };
+
+	/** @type {string[]} */
+	const svelte_extensions = config.extensions ?? ['.svelte'];
+	const extensions = [...svelte_extensions, '.ts', '.js'];
+	// TODO read tsconfig/jsconfig if available? src/** will be good for 99% of cases
+	const files = glob('src/**', { filesOnly: true, dot: true }).map((file) =>
+		file.replace(/\\/g, '/')
+	);
+
+	for (const file of files) {
+		if (extensions.some((ext) => file.endsWith(ext))) {
+			if (svelte_extensions.some((ext) => file.endsWith(ext))) {
+				update_svelte_file(file);
```

Comment:
this would break with mdsvex `.svx`  , safer to only do .svelte by default and maybe ask the user for other extensions that contain svelte code that is compatible with the basic regex used in update_svelte_file

Reply:
I commented that out because I think it's going to do more harm than good since `.svx` is probably the most common usage of it